packages/common-ts/src/base-service/base-service-v2.ts

```diff
@@ -52,6 +54,18 @@ export type MetricsSpec<TMetrics extends MetricsV2> = {
 
 export type ExpressRouter = Router
 
+// need to make an example to infer the type
+class ExampleTRPCApi<
+  TMetrics extends MetricsV2,
+  TRoutes extends TrpcApi<TMetrics>['routes']
+> extends TrpcApi<TMetrics> {
+  routes: TRoutes = this.router({}) as any
+}
+type ExtendedTrpcApi<
+  TMetrics extends MetricsV2,
+  TRoutes
+> = typeof ExampleTRPCApi<TMetrics, TRoutes>
+
```

Comment:
This was the quickest way I could get the right type.   I need a type that is `class that extends TrpcApi`.    There is likely a better way to do this in typescript but this was easy and works